service/policy/db/attribute_values.go

```diff
@@ -474,59 +222,42 @@ func (c PolicyDBClient) UnsafeDeleteAttributeValue(ctx context.Context, toDelete
 		return nil, fmt.Errorf("fqn mismatch [%s]: %w", fqn, db.ErrNotFound)
 	}
 
-	sql, args, err := unsafeDeleteAttributeValueSQL(id)
+	count, err := c.Queries.DeleteAttributeValue(ctx, id)
 	if err != nil {
-		return nil, err
+		return nil, db.WrapIfKnownInvalidQueryErr(err)
 	}
-
-	if err := c.Exec(ctx, sql, args); err != nil {
-		return nil, err
+	if count == 0 {
+		return nil, db.ErrNotFound
 	}
 
 	return &policy.Value{
 		Id: id,
 	}, nil
-}
 
-func assignKeyAccessServerToValueSQL(valueID, keyAccessServerID string) (string, []interface{}, error) {
-	t := Tables.AttributeValueKeyAccessGrants
-	return db.NewStatementBuilder().
-		Insert(t.Name()).
-		Columns("attribute_value_id", "key_access_server_id").
-		Values(valueID, keyAccessServerID).
-		ToSql()
 }
 
 func (c PolicyDBClient) AssignKeyAccessServerToValue(ctx context.Context, k *attributes.ValueKeyAccessServer) (*attributes.ValueKeyAccessServer, error) {
-	sql, args, err := assignKeyAccessServerToValueSQL(k.GetValueId(), k.GetKeyAccessServerId())
+	_, err := c.Queries.AssignKeyAccessServerToAttributeValue(ctx, AssignKeyAccessServerToAttributeValueParams{
+		AttributeValueID:  k.GetValueId(),
```

Comment:
This was the existing behavior, but it's interesting that we're not validating in the protos that these fields are provided. The query itself will fail, though, because the column type is UUID.